### DISCUSSION TITLE
Invalid response for ReadDTC command

### DISCUSSION THREAD

**Discussion by AndyIreland** [cedricp/ddt4all#1029] [Q&A]
anyone know why I am getting this error .... But only on one of the BCM/UCH ECU on my car (Renault Zoe 2013 EV) 
![image](https://github.com/cedricp/ddt4all/assets/49172217/7ffb405c-8a7d-4e80-8a7b-7f31284401b9)

LOGS:
[ecu_ddt.txt](https://github.com/cedricp/ddt4all/files/14312053/ecu_ddt.txt)
[elm_ddt.txt](https://github.com/cedricp/ddt4all/files/14312054/elm_ddt.txt)
[screens.txt](https://github.com/cedricp/ddt4all/files/14312055/screens.txt)

**Comment by AndyIreland** [comment]
on other ECU's I manage to put to extended diagnostics mode: 
![image](https://github.com/cedricp/ddt4all/assets/49172217/5a434421-1f12-4582-a7ed-5e63b3e7fad8)

but on the BCM/UCH it does not show that it just defaults to After sales:
![image](https://github.com/cedricp/ddt4all/assets/49172217/ccbc42d2-a327-47f9-b604-2bd865f8b1a2)

Could it be because of that?

**Comment by KarelSvo** [comment]
############################################################
#connect to: b'BCM95_SW25_V1.1' Addr:26
############################################################
>[17:45:01.761]AT SH 745
<[0.011]AT SH 745
OK

>
>[17:45:01.773]AT CRA 765
<[0.013]AT CRA 765
OK

>
>[17:45:01.786]AT FC SH 745
<[0.014]AT FC SH 745
OK

>
>[17:45:01.800]AT FC SD 30 00 00
<[0.015]AT FC SD 30 00 00
OK

>
>[17:45:01.815]AT FC SM 1
<[0.012]AT FC SM 1
OK

>
>[17:45:01.828]AT SP 6
<[0.015]AT SP 6
OK

>
>[17:45:01.846]0210C0
<[0.215]0210C0
0250C0FFFFFFFFFF

>
>[17:45:02.809]023E01
<[0.24]023E01
017EFFFFFFFFFFFF

>
>[17:45:04.311]023E01
<[0.236]023E01
017EFFFFFFFFFFFF

>
>[17:45:05.801]023E01
<[0.22]023E01
017EFFFFFFFFFFFF

>
>[17:45:07.309]023E01
<[0.087]023E01
017EFFFFFFFFFFFF

>
>[17:45:08.799]023E01
<[0.097]023E01
017EFFFFFFFFFFFF

>
>[17:45:10.305]023E01
<[0.091]023E01
017EFFFFFFFFFFFF

>
>[17:45:11.803]023E01
<[0.096]023E01
017EFFFFFFFFFFFF

>
>[17:45:12.125]0210C0
<[0.098]0210C0
0250C0FFFFFFFFFF

>
>[17:45:12.222]031902FF
<[0.157]031902FF
11A75902B995F112
21109532121095F1
2211109532111095
2352821095E39310
2495E5931095E793
251095E993105A84
2631105A8496105A
278531105A859610
285A8331105A8396
29105A8231105A82
2A961095B0021095
2B61231095602410
2C95602310953E14
2D10953F14109540
2E1410953B141095
2F3C1410953D1410
209550011095514A
211095F
BUFFER FULL

Chinese Elm interface at the performance limit, because a lot of error codes respond to 1902FF.

**Reply by Furtif** [reply]
How do they have so many problems? I think it's worth changing cars ... oufff

**Comment by KarelSvo** [comment]
There are a large number of phantom errors. The UCH was taken from the Megane III. Pyren is more suitable for fault diagnosis. The filter function of the CLIP Database only shows the error codes relevant to the ZOE.



> but on the BCM/UCH it does not show that it just defaults to After sales:

10C0 is 10C0 start of the diagnostic session, no matter what name the ddt file gives it.
Newer Ecus start the diagnostic session with 1003.
After sales (default) [10C0] is always displayed in the list because it is specified by ddt4all https://github.com/cedricp/ddt4all/blob/master/parameters.py#L821
The other options from the list come from the currently open ddt file.

**Reply by AndyIreland** [reply]
so what does it mean that when I re-scan for errors in PYREN and everything on other ecu's is 0 (meaning no DTC errors stored) but when i look at the BCM/UCH it says 4 errors. When I press 1 then there are no errors listed and just 'clear errors' underneath - if I do that and re-scan for errors it still says 4 errors. They just wont clear whatever I do! 
So if I go to DDT4ALL to clear DTC errors DDT4ALL throws up the error message and I cannot clear it that way either - whats next ? - it un-nerves me to see 4 errors where every other ECU on the car is error free.

**Comment by KarelSvo** [comment]
Without Pyren log data it's just a guessing game.
